monai/transforms/spatial/functional.py

```diff
@@ -160,3 +167,270 @@ def spatial_resample(
+        "dtype": str(dtype)[6:],  # dtype as string; remove "torch": torch.float32 -> float32
+        "new_dim": len(orig_size) - input_ndim,
+    }
+    meta_info = TraceableTransform.track_transform_meta(
+        img,
+        sp_size=out_size,
+        affine=scale_affine(orig_size, out_size),
+        extra_info=extra_info,
+        orig_size=orig_size,
+        transform_info=transform_info,
+        lazy_evaluation=transform_info.get(TraceKeys.LAZY_EVALUATION, False),
+    )
+    out = convert_to_tensor(img.as_tensor() if isinstance(img, MetaTensor) else img, track_meta=get_track_meta())
+    if transform_info.get(TraceKeys.LAZY_EVALUATION, False) or tuple(convert_to_numpy(orig_size)) == out_size:
+        if anti_aliasing:
+            warnings.warn("anti-aliasing is not compatible with lazy evaluation.")
+        return out.copy_meta_from(meta_info) if isinstance(out, MetaTensor) else meta_info
+    img_ = convert_to_tensor(out, dtype=dtype, track_meta=False)  # convert to a regular tensor
+    if anti_aliasing and any(x < y for x, y in zip(out_size, img_.shape[1:])):
+        factors = torch.div(torch.Tensor(list(img_.shape[1:])), torch.Tensor(out_size))
+        if anti_aliasing_sigma is None:
+            # if sigma is not given, use the default sigma in skimage.transform.resize
+            anti_aliasing_sigma = torch.maximum(torch.zeros(factors.shape), (factors - 1) / 2).tolist()
+        else:
+            # if sigma is given, use the given value for downsampling axis
+            anti_aliasing_sigma = list(ensure_tuple_rep(anti_aliasing_sigma, len(out_size)))
+            for axis in range(len(out_size)):
+                anti_aliasing_sigma[axis] = anti_aliasing_sigma[axis] * int(factors[axis] > 1)
+        anti_aliasing_filter = GaussianSmooth(sigma=anti_aliasing_sigma)
+        img_ = convert_to_tensor(anti_aliasing_filter(img_), track_meta=False)
+    resized = torch.nn.functional.interpolate(
+        input=img_.unsqueeze(0), size=out_size, mode=mode, align_corners=align_corners
+    )
+    out, *_ = convert_to_dst_type(resized.squeeze(0), out, dtype=torch.float32)
+    return out.copy_meta_from(meta_info) if isinstance(out, MetaTensor) else out
+
+
+def rotate(img, angle, output_shape, mode, padding_mode, align_corners, dtype, transform_info):
+    im_shape = img.peek_pending_shape() if isinstance(img, MetaTensor) else img.shape[1:]
+    input_ndim = len(im_shape)
+    if input_ndim not in (2, 3):
+        raise ValueError(f"Unsupported image dimension: {input_ndim}, available options are [2, 3].")
+    _angle = ensure_tuple_rep(angle, 1 if input_ndim == 2 else 3)
+    transform = create_rotate(input_ndim, _angle)
+    if output_shape is None:
+        corners = np.asarray(np.meshgrid(*[(0, dim) for dim in im_shape], indexing="ij")).reshape((len(im_shape), -1))
+        corners = transform[:-1, :-1] @ corners  # type: ignore
+        output_shape = corners.ptp(axis=1) + 0.5
```

Comment:
May still need `output_shape = np.asarray(corners.ptp(axis=1) + 0.5, dtype=int)` to ensure the type, otherwise we may get float values in `peek_pending_shape`. I will address this issue in: https://github.com/Project-MONAI/MONAI/pull/6080

Reply:
Sure, please go ahead and fix it. It’s a good point, maybe we don’t need integer shapes during lazy evaluations, we just cast them into integers when resampling, I’ll think about it.